BioForm/profiles/backends.py

```diff
@@ -0,0 +1,25 @@
+from django.contrib.auth.backends import ModelBackend
+from django.contrib.auth import get_user_model
+
+from profiles.models import Account, AccountManager
+from django.contrib import messages
+
+
+class AccountBackend(ModelBackend):
+
+    def authenticate(self, request, username = None, password = None, **kwargs):
+        UserModel = get_user_model
+
+        if username is None:
+            username = kwargs.get(UserModel.USERNAME_FIELD)
+        
+        try:
+            case_sensitive_email = AccountManager.get_by_natural_key(email__iexact=username)
```

Comment:
What does `get_by_natural_key` is used for?
Why are you using a username to match with email?

Reply:
1) get_by_natural_key is a function I overrided in AccountManager that is used to get a user Instance based on the USERNAME_FIELD. I overrided the USERNAME_FIELD to take email as to authenticate using email and not username.

2) In the Django User Model (which I am overriding to make a custom user model) the authentication is done with the keyfield username. Even if we override this by changing USERNAME_FIELD to email, we still have to keep the name of the variable username as that is how it has been coded in the backend. I searched about this and this is a flaw in Django which makes overriding this confusing but it can be done so I tired it.

Reply:
Not using backends.py anymore.